### PR TITLE
Update create-player-hazards.md

### PR DESCRIPTION
**Pull request by TencoDev**
## Issue
The current script includes a fail-safe for cases where the `Players.PlayerAdded` event might not work perfectly using a for-loop for existing players. If a player collects coins before the `Players.PlayerAdded` event is executed or before the 'for loop' executes, their coin count would be reset to zero. This pull request addresses this issue by introducing a check to prevent resetting the count if the player already has coins.

## Changes
The code change involves modifying the `onPlayerAdded` function to check if the player already has coins (`currentCoins`). If `currentCoins` is `nil or zero`, it returns zero. This ensures that a player's coin count is only reset if they don't have any coins collected.

Code Change Illustration:
```lua
-- Before
for _, player in ipairs(Players:GetPlayers()) do
	onPlayerAdded(player)
end
    -- Resets the coin count to zero without checking for current coins as the method 'onPlayerAdded' does not check for currentCoins

-- After
local function onPlayerAdded(player)
	-- Check if the player already has coins
	local currentCoins = PlayerData.getValue(player, COIN_KEY_NAME)
	if currentCoins == nil or currentCoins == 0 then
		-- Reset player coins to 0 if they don't have any coins
		updatePlayerCoins(player, function(_)
			return 0
		end)
	end
```

Since the 'for loop' invokes the onPlayerAdded method, the fix was implemented there. Alternatively, it can be written in the for loop code block.

## Checks

By submitting your pull request for review, you agree to the following:

- [x] This contribution was created in whole or in part by me, and I have the right to submit it under the terms of this repository's open source licenses.
- [x] I understand and agree that this contribution and a record of it are public, maintained indefinitely, and may be redistributed under the terms of this repository's open source licenses.
- [x] To the best of my knowledge, all proposed changes are accurate.

### CODE REVIEW

**Comment by aetter** [comment]
Hi @TencoDev, thanks a bunch for sending this over. Even though there's some risk of a player gathering coins over a short period of time and having them reset, I think it's a fair tradeoff for an experience that's designed for brand new creators to learn from. Strategies like freezing a player until loading finishes or code scheduling seems like a more advanced topic that we'd want to skirt around in a beginner's tutorial.

If you're interested in adding a sentence about this potential problem to the code explanation, I think that could be a great addition, but I'd prefer to the leave the code as-is.

File: content/en-us/tutorials/core/scripting/create-player-hazards.md [hunk 1]
```diff
@@ -152,6 +153,7 @@ lifecycle events:
   can execute.
 - `Class.Player.PlayerRemoved` fires when a player leaves the experience to
   clean up player state.
+- In the current setup, there is a potential issue if a player collects coins before the `Players.PlayerAdded` event is executed or before the `for loop` executes. In such cases, the player's coin count could be reset to zero upon character creation.To mitigate this issue, consider implementing more advanced techniques, such as freezing the player's character until the necessary initialization is complete, or using code scheduling to ensure a seamless experience. However, these solutions involve more complex scripting concepts and are beyond the scope of this beginner's tutorial.
```
Comment (aetter) [inline content/en-us/tutorials/core/scripting/create-player-hazards.md:156]:
```suggestion
- This code contains a potential issue where players could collect coins before the `Players.PlayerAdded` event executes and then have their coin counts reset to zero. To mitigate this issue, consider solutions such as code scheduling or freezing the player's character until initialization finishes. However, these solutions involve more complex scripting concepts that are beyond the scope of this tutorial.
```

**Review by aetter — APPROVED** [review]

**Comment by aetter** [comment]
Hey @TencoDev, thanks a bunch. I made a few wording changes, but this is ready to go after you do the legal checks (I re-added them to the PR summary).

**Comment by TencoDev** [comment]
@aetter Sorry for the delayed response my friend, Everything is perfect.

**Comment by aetter** [comment]
Hey @TencoDev, thanks a bunch for checking things out. I can't merge until you check the boxes in your [initial comment here](https://github.com/Roblox/creator-docs/pull/179#issue-1932062430).

Screenshot 2023-10-23 at 8 18 28 AM

**Comment by TencoDev** [comment]
@aetter Sorry for the oversight, I have checked the boxes.

**Comment by aetter** [comment]
No sweat at all, @TencoDev, thank you!

**Pull request merged**